app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
```

Comment:
why not do `$(':not[.has-explicit]').prop(disabled, true)` or something of the sort in `serializeForm` before calling `$('form').serialize()` instead of creating this complex filter on all the form elements? Or even better, simply disable the relevant inputs when they need to be disabled rather then doing it before the form submission. `.serialize()` by default ignores any disabled inputs.